### PR TITLE
Add ref flags for function params/locals to AOT module instead of committing them in GC AOT code

### PR DESCRIPTION
**Pull request by TianlongLiang**

### CODE REVIEW

File: core/iwasm/aot/aot_runtime.h [hunk 1]
```diff
@@ -183,6 +190,11 @@ typedef struct AOTModule {
     uint32 *max_stack_cell_nums;
 #endif
 
+#if WASM_ENABLE_GC != 0
+    /* params + locals ref flags of (both import and AOTed) functions */
+    struct LocalRefFlag *func_local_ref_flags;
```
Comment (wenyongh) [inline core/iwasm/aot/aot_runtime.h:195]: Remove `struct`

File: core/iwasm/aot/aot_loader.c [hunk 2]
```diff
@@ -2574,6 +2574,41 @@ load_function_section(const uint8 *buf, const uint8 *buf_end, AOTModule *module,
 #endif
     }
 
+#if WASM_ENABLE_GC != 0
+    size = sizeof(LocalRefFlag)
```
Comment (wenyongh) [inline core/iwasm/aot/aot_loader.c:2578]: Had better add comment for this code piece.

File: core/iwasm/aot/aot_runtime.h [hunk 3]
```diff
@@ -362,7 +374,9 @@ typedef struct AOTFrame {
      *  local area: parameters and local variables
      *  stack area: wasm operand stack
      *  frame ref flags (GC only):
-     *      whether each cell in local and stack area is a GC obj
+     *      whether each cell in local(LLVM JIT) and stack area is a GC obj
+     *                           paddings(AOT)
+     *                           actual ref flags for local are in AOT module
```
Comment (wenyongh) [inline core/iwasm/aot/aot_runtime.h:379]: Here AOTFrame is only for AOT, JIT's frame is WASMInterpFrame.
How about below like:
```C
     *  frame ref flags (GC only):
     *      whether each cell in local and stack area is a GC obj
     *      currently local's ref flags are stored in AOTModule,
     *      here we only reserve the padding bytes 
```

File: core/iwasm/compilation/aot_compiler.c [hunk 4]
```diff
@@ -365,6 +366,9 @@ aot_gen_commit_values(AOTCompFrame *frame)
                             (p + 2)->committed_ref = (p + 3)->committed_ref =
                                 p->ref + 1;
                     }
+                    p++;
+                    p++;
+                    p++;
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_compiler.c:371]: How about `p += 3`

File: core/iwasm/compilation/aot_compiler.c [hunk 5]
```diff
@@ -318,11 +318,11 @@ aot_gen_commit_values(AOTCompFrame *frame)
     LLVMValueRef value;
     uint32 n;
 
-    for (p = frame->lp; p < frame->sp; p++) {
+    /* First, commit reference flags, ignore local(params + locals) ref flags */
+    for (p = frame->lp + frame->max_local_cell_num; p < frame->sp; p++) {
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_compiler.c:322]: Should not ignore local ref flags for JIT? Not sure why JIT's CI runs good.
For JIT, should be `p = frame->lp`.

**Review by xujuntwt95329 — COMMENTED** [review]
LGTM

**Pull request merged**